exporter/attestation/filter.go

```diff
@@ -1,45 +1,49 @@
 package attestation
 
 import (
-	"bytes"
+	"strconv"
 
 	"github.com/moby/buildkit/exporter"
+	"github.com/moby/buildkit/solver/result"
 )
 
-func Filter(attestations []exporter.Attestation, include map[string][]byte, exclude map[string][]byte) []exporter.Attestation {
-	if len(include) == 0 && len(exclude) == 0 {
-		return attestations
-	}
-
-	result := []exporter.Attestation{}
+func FilterInline(attestations []exporter.Attestation) (matching []exporter.Attestation, nonMatching []exporter.Attestation) {
 	for _, att := range attestations {
-		meta := att.Metadata
-		if meta == nil {
-			meta = map[string][]byte{}
-		}
-
-		match := true
-		for k, v := range include {
-			if !bytes.Equal(meta[k], v) {
-				match = false
-				break
+		v, ok := att.Metadata[result.AttestationInlineOnlyKey]
+		if ok {
+			b, err := strconv.ParseBool(string(v))
+			if b && err == nil {
+				matching = append(matching, att)
+				continue
 			}
 		}
-		if !match {
-			continue
-		}
+		nonMatching = append(nonMatching, att)
+	}
+	return matching, nonMatching
+}
 
-		for k, v := range exclude {
-			if bytes.Equal(meta[k], v) {
-				match = false
-				break
+func FilterReasons(attestations []exporter.Attestation, reasons []string) (matching []exporter.Attestation, nonMatching []exporter.Attestation) {
+	if reasons == nil {
+		// don't filter if no filter provided
+		return attestations, nil
+	}
+
+	for _, att := range attestations {
+		target, ok := att.Metadata[result.AttestationReasonKey]
```

Comment:
Not from this PR but I don't understand why this is called a "reason". It is attestation type. So why can't we just use the already existing predicate type field?